homeassistant/components/http/auth.py

```diff
@@ -67,6 +67,8 @@ def async_sign_path(
         secret,
         algorithm="HS256",
     )
+    if "?" in path
```

Comment:
- Use `yarl.URL` for mutating URLs
- We need to decide if the auth signature now also need to verify that the passed query params are like at the time of signing. 
- Tests, tests, tests